compiler/rustc_data_structures/src/sync.rs

```diff
@@ -339,7 +339,10 @@ cfg_if! {
             t: T,
             for_each: impl Fn(T::Item) + Sync + Send,
         ) {
-            t.into_par_iter().for_each(for_each)
+            let ps: Vec<_> = t.into_par_iter().map(|i| catch_unwind(AssertUnwindSafe(|| for_each(i)))).collect();
```

Comment:
Does rayon support `.collect::<Result<Vec<()>, Box<dyn Any + Send>>()`? That would be faster than iterating over all elements to check if any is `Err` I think.

Reply:
We indeed need to iterate over all elements, not just know if any of them is `Err`. Because the err message will be output during the iteration